src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/BlazorDebug/TerminateDebugHandler.ts

```diff
@@ -10,6 +10,17 @@ import { RazorLogger } from '../RazorLogger';
 
 import { HOSTED_APP_NAME, JS_DEBUG_NAME } from './Constants';
 
+function processExists(pid: number): boolean {
+  try {
+    // Executing a kill with a '0' signal will not terminate
+    // the process but check if it exists
+    process.kill(pid, 0);
```

Comment:
So if it does exist it no-ops? I find it interesting if it doesn't exist it throws an error

Reply:
Correct. The command runs through the steps that would typically be taken when sending a kill signal, but doesn't actually send the signal. That way, it effectively allows you to check if the process exists and if it can be operated on.